### DISCUSSION TITLE
Is multiple hosts in the connection string supported?

### DISCUSSION THREAD

**Discussion by steven-xufan** [psycopg/psycopg#524] [Q&A] [answered]
Hi, I'm trying to specify multiple hosts in the connection string (as described in [libpq doc]( https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-MULTIPLE-HOSTS)) so that it can automatically connect to another host in case one is not available. However, it does not seem to work in psycopg(3). I tried this in psycopg2 and it was working. I checked this in macOS and python:3.10.6-slim docker image. I also did a search in the [Issues page](https://github.com/psycopg/psycopg/issues) and [Doc page](https://www.psycopg.org/psycopg3/docs/index.html), but could not find anything related. Is this feature supported?

To replicate the behavior:
```python
import psycopg

# first pair of host/port does not exist. it should connect to the second pair.
con = psycopg.connect("postgresql://postgres:postgres@localhost:5442,localhost:5432/test")
# the above line fails with `psycopg.OperationalError: connection failed: ...`

cur = con.cursor()
cur.execute("select version()")
print(cur.fetchone())

# for psycopg2, just replace the module name.
```

Environment:
- OS: macOS 12.6.1, python:3.10.6-slim (docker)
- psycopg(3) version: psycopg-binary:3.1.8
- psycopg2 version: psycopg2-binary:2.9.5
- postgresql version: 14.1
- python version: 3.10.6
- pip version: 22.2.1

**Comment by dvarrazzo** [comment, marked as answer]
What is the libpq that gets installed? You can check it from [`psycopg.pq.version()`](https://www.psycopg.org/psycopg3/docs/api/pq.html#psycopg.pq.version)

**Reply by steven-xufan** [reply]
on my local machine (macOS), libpq version is `140006`. in the docker image, libpq version is `150001`.

**Reply by dvarrazzo** [reply]
Both versions support multiple addresses. pg3 works for me:
```python
>>> import psycopg

>>> psycopg.connect("postgresql://localhost:5430/psycopg3_test")  # nothing listening here
...
OperationalError: connection failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

>>> psycopg.connect("postgresql://localhost:5430,localhost:5432/psycopg3_test")
<psycopg.Connection [IDLE] (host=localhost user=piro database=psycopg3_test) at 0x7f3717c93490>

>>> psycopg.pq.version()
150002
```

**Reply by steven-xufan** [reply]
Thanks for getting back. I managed to verify that this is working. 

It somehow did not work if i'm trying to reach a random `port` on the mac. Sorry for the trouble.
